nautobot/core/api/parsers.py

```diff
@@ -103,6 +105,34 @@ def insert_nested_dict(keys, value, current_dict):
 
         return result_dict
 
+    def _remove_object_not_found_values(self, data):
+        """Remove all `ObjectNotFound` field lookups from the given data, and swap out 'NaN' and
+        'ObjectNotFound' values for `None`.
+
+        If all the lookups for a field are 'ObjectNotFound', it indicates that the field does not exist,
+        and it needs to be removed to prevent unnecessary database queries.
+
+        Args:
+            data (dict): A dictionary containing field natural key lookups and their corresponding values.
+
+        Returns:
+            dict: A modified dictionary with 'ObjectNotFound' values removed, and 'NaN' and 'ObjectNotFound' swapped for `None`.
+        """
+        grouped_data = {}
+        for lookup, lookup_value in data.items():
+            field_name = lookup.split("__", 1)[0]
+            grouped_data.setdefault(field_name, {}).update({lookup: lookup_value})
+
+        valid_data = {}
+
+        for lookup_group in grouped_data.values():
+            for lookup, lookup_value in lookup_group.items():
+                if any(value != CSV_OBJECT_NOT_FOUND for value in lookup_group.values()):
```

Comment:
Is this right? After iterating on `lookup_group.items()` you're still iterating over `lookup_group.values()` for each item? I assume it's right, but can you please add some commenting for this block on why you need to do this? It's not clear from the block of code exactly why. Thank you.